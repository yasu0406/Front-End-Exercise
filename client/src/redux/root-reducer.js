import { combineReducers } from 'redux';
import communitiesReducer from './communities/communities.reducer';

const rootReducer = combineReducers({
    communities: communitiesReducer
});

export default rootReducer;