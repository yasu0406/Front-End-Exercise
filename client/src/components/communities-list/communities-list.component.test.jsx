import React from 'react';
import { mount } from 'enzyme';
import { store } from '../../redux/store';
import { Provider } from 'react-redux';
import CommunitiesList from './communities-list.component';

describe('CommunitiesList', () => {
    let wrapper;
    let mockFetchCommunities;
    let mockFetchCommunitiesAverage;

    beforeEach(() => {
        mockFetchCommunities = jest.fn();
        mockFetchCommunitiesAverage = jest.fn();
        const mockProps = {
            fetchCommunities: mockFetchCommunities,
            fetchCommunitiesAverage: mockFetchCommunitiesAverage
        };
        wrapper = mount(
            <Provider store={store}>
                <CommunitiesList {...mockProps} />
            </Provider>
        );
    });
    it('should render CommunitiesList component', () => {
        expect(wrapper).toMatchSnapshot();
    });
});