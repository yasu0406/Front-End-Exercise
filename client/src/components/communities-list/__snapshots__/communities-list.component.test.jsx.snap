// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommunitiesList should render CommunitiesList component 1`] = `ReactWrapper {}`;
